src/script-events.c

```diff
@@ -88,89 +291,51 @@ int kretprobe__ret_script_load(struct pt_regs *ctx)
         return 0;
     }
 
+    // Create the initial event in the cache
+    bpf_printk("Creating the initial event\n");
+    push_telemetry_event(ctx, &ev);
+
     // Get filename
-    ret = read_value(bprm, CRC_BPRM_FILENAME, &file_ptr, sizeof(file_ptr));
+    ret = read_value(bprm, CRC_BPRM_FILENAME, &str, sizeof(str));
     if (ret < 0)
     {
         return 0;
     }
 
     // Read the filename
-    bpf_probe_read_str(&ev.u.script_info.path, sizeof(ev.u.script_info.path), file_ptr);
+    count = bpf_probe_read_str(&ev.u.v.value, sizeof(ev.u.v.value), str);
+    str += sizeof(ev.u.v.value) - 1;
+    bpf_printk("Count: %d\n", count);
+    br = ev.u.v.value[0];
 
     // Output the path
-    bpf_perf_event_output(ctx, &script_events, bpf_get_smp_processor_id(), &ev, sizeof(ev));
-
-    // If the path is / then we don't need to do anything else. This is the case when you run
-    // a command that is in your PATH or you run a script using the absolute path. If you run
-    // a script from a local directory (i.e. ./myscript.sh) then the event that gets output
-    // here is ./myscript.sh and then the later logic find the current working directory, appends
-    // that to the path here and uses that to get the absolute path to the script. It needs that
-    // path so it can read the file contents
-    end = ev.u.script_info.path[0];
-    if (end == '/')
+    ev.id = id;
+    ev.done = 0;
+    ev.telemetry_type = TE_CHAR_STR;
+    push_telemetry_event(ctx, &ev);
+    if (count >= sizeof(ev.u.v.value))
     {
-        return 0;
-    }
+        bpf_printk("count == to value size. Reading more...\n");
+        READ_CHAR_STR_N(10)
 
-    u64 offset = 0;
-    void *ptr = (void *)bpf_get_current_task();
-
-    // Get the pointer to the fs field in current
-    if (read_value(ptr, CRC_TASK_STRUCT_FS, &ptr, sizeof(ptr)) < 0)
-    {
-        goto Skip;
+    DONE_READING:
+        // If after we have read the full path the first chararcter was a /
+        // then we are done
+        if (br == '/')
+        {
+            goto Skip;
+        }
     }
-
-    // Get the offset for pwd in fs_struct
-    SET_OFFSET(CRC_FS_STRUCT_PWD);
-
-    // Read the Dentry pointer for pwd
-    ptr = ptr + *(u32 *)offset; // ptr to pwd
-    if (read_value(ptr, CRC_PATH_DENTRY, &ptr, sizeof(ptr)) < 0)
+    else if (br == '/')
     {
+        bpf_printk("Count < value size. goto Skip\n");
```

Comment:
`:%s/bpf_printk\(".+"\)\;//gc`

Reply:
Yea sorry. I committed these changes last not forgetting they were going to update this PR